### PR TITLE
[CARBONDATA-1088] Minimize the driver side block index.

### PR DESCRIPTION
**Pull request by ravipesala**

### CODE REVIEW

**Comment by ravipesala** [comment]
retest this please

**Comment by ravipesala** [comment]
retest this please

File: core/src/main/java/org/apache/carbondata/core/datastore/impl/array/BlockArrayIndexBuilder.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import org.apache.carbondata.core.memory.MemoryAllocator;
+import org.apache.carbondata.core.memory.MemoryAllocatorFactory;
+import org.apache.carbondata.core.memory.MemoryBlock;
+import org.apache.carbondata.core.metadata.blocklet.DataFileFooter;
+import org.apache.carbondata.core.metadata.blocklet.index.BlockletMinMaxIndex;
+
+/**
+ * Builder for storing block index in unsafe array format
+ */
+public class BlockArrayIndexBuilder implements BtreeBuilder {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/impl/array/BlockArrayIndexBuilder.java]: We should use interface defined by #958 
Suggest to merge #958 first then rebase on it, ok?

**Comment by ravipesala** [comment]
Closed as this is refactored in new PR https://github.com/apache/carbondata/pull/958

**Pull request closed without merging**